### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Ignat1990**

### CODE REVIEW

File: src/main/java/bean/Orders.java [hunk 1]
```diff
@@ -20,12 +20,9 @@ public class Orders {
 
 
     @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "users_id")
+    @JoinColumn(name = "id")
     private Users users;
 
-    @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "status_orders_id")
-    private StatusOrder statusOrder;
```
Comment (DHrytsuk) [inline src/main/java/bean/Orders.java:28]: Why did you remove it?

File: src/main/java/service/BasketService.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import bean.Basket;
+import java.util.List;
+
+
+
+    public interface BasketService {
+        public List<Basket> getAllBasket();
+        public void deleteBasket(Long id);
+        public Basket addBookInBasket(Basket basket);
+        public Basket getBasket(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/BasketService.java]: rename method to `find`

File: src/main/java/service/BasketService.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package service;
+import bean.Basket;
+import java.util.List;
+
+
+
+    public interface BasketService {
+        public List<Basket> getAllBasket();
```
Comment (DHrytsuk) [inline src/main/java/service/BasketService.java]: rename method to `findAll`

File: src/main/java/service/BasketService.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        public void deleteBasket(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/BasketService.java]: rename to `delete`

File: src/main/java/service/BookService.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package service;
+
+import bean.Book;
+import bean.Orders;
+import java.util.List;
+
+public interface BookService {
+    public List<Book> getAllBooks();
```
Comment (DHrytsuk) [inline src/main/java/service/BookService.java:8]: rename to `findAll`

File: src/main/java/service/BookService.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    public Book getBookById(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/BookService.java:9]: rename to `find`

File: src/main/java/service/BookService.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    public void saveBook(Book book);
```
Comment (DHrytsuk) [inline src/main/java/service/BookService.java:10]: rename to `save`

File: src/main/java/service/BookService.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    public void deleteBook(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/BookService.java:11]: rename to `delete`

File: src/main/java/service/BasketService.java (hunk 3 above)
Comment (DHrytsuk) [inline src/main/java/service/BasketService.java]: No need to add `public` in interface methods

File: src/main/java/service/OrdersService.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+package service;
+
+import bean.Orders;
+
+
+import java.util.List;
+
+public interface OrdersService {
+    public Orders saveOrder(Orders order);
```
Comment (DHrytsuk) [inline src/main/java/service/OrdersService.java:9]: rename to `save`

File: src/main/java/service/OrdersService.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+    public Orders findById(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/OrdersService.java:11]: rename to `find`

File: src/main/java/service/OrdersService.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+    public List<Orders> findAllOrder();
```
Comment (DHrytsuk) [inline src/main/java/service/OrdersService.java:13]: rename to `findAll`

File: src/main/java/service/OrdersService.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+    public void deleteOrder(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/OrdersService.java:15]: rename to `delete`

File: src/main/java/service/OrdersService.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+    public Orders updateOrder(Orders order);
```
Comment (DHrytsuk) [inline src/main/java/service/OrdersService.java:17]: rename to `update`

File: src/main/java/service/UsersService.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
+package service;
+
+import bean.Orders;
+import bean.Users;
+import java.util.List;
+
+public interface UsersService {
+    public Users findById(Long id);
```
Comment (DHrytsuk) [inline src/main/java/service/UsersService.java:8]: rename to `find`

File: src/main/java/service/UsersService.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+    public List<Users> findAllUsers();
```
Comment (DHrytsuk) [inline src/main/java/service/UsersService.java:9]: rename to `findAll`

File: src/main/java/service/UsersService.java [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+    public void saveUser(Users user);
```
Comment (DHrytsuk) [inline src/main/java/service/UsersService.java:10]: rename to `save`

File: src/main/java/service/UsersService.java [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+    public void deleteUser(long id);
```
Comment (DHrytsuk) [inline src/main/java/service/UsersService.java:11]: rename to `delete`

File: src/main/java/service/lmpl/BasketServiceImpl.java [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+import service.BasketService;
+
+import java.util.List;
+
+import static java.awt.Container.log;
+
+@Service
+public class BasketServiceImpl implements BasketService {
+
+    final
```
Comment (DHrytsuk) [inline src/main/java/service/lmpl/BasketServiceImpl.java:18]: remove final and add access modifier (private or protected)

File: src/main/java/service/lmpl/BasketServiceImpl.java [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+    BookRepository bookRepository;
+    final
+    BasketRepository basketRepository;
+
+    public BasketServiceImpl(BookRepository bookRepository, BasketRepository basketRepository) {
```
Comment (DHrytsuk) [inline src/main/java/service/lmpl/BasketServiceImpl.java:23]: Use @Autowired annotation for fields instead of constructor

File: src/main/java/service/lmpl/UserDetailsServiceImpl.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import repository.RolesRepository;
+
+public class UserDetailsServiceImpl {
+
+protected RolesRepository rolesRepository;
+
+@Autowired
```
Comment (DHrytsuk) [inline src/main/java/service/lmpl/UserDetailsServiceImpl.java:11]: It seems, it should be @Component

File: src/main/java/service/lmpl/BasketServiceImpl.java (hunk 19 above)
Comment (DHrytsuk) [reply src/main/java/service/lmpl/BasketServiceImpl.java:23]: same in other places

File: src/main/java/service/lmpl/BasketServiceImpl.java (hunk 18 above)
Comment (DHrytsuk) [reply src/main/java/service/lmpl/BasketServiceImpl.java:18]: same in other places

File: src/main/java/bean/Orders.java (hunk 1 above)
Comment (Ignat1990) [reply src/main/java/bean/Orders.java:28]: Because have problem with idea, has exception!
